### PR TITLE
refactor(components): [tabs] improve types

### PR DESCRIPTION
**Pull request by tolking**
Please make sure these boxes are checked before submitting your PR, thank you!

- [x] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [x] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

### CODE REVIEW

**Review by sxzz — DISMISSED** [review]

**Review by sxzz — COMMENTED** [review]
There's a problem with `onTabClick`. I think we should move it to `emits`. Otherwise, TS typing is wrong.

**Review by holazz — APPROVED** [review]

**Review by sxzz — COMMENTED** [review]
Why add click event to icon?

**Comment by holazz** [comment]
> Why add click event to icon?

Maybe to fix this?

https://github.com/element-plus/element-plus/blob/d0dbd19f75248adb4e56755a02c376a33f83f039/packages/components/tabs/src/tab-nav.tsx#L320

**Comment by tolking** [comment]
> https://github.com/element-plus/element-plus/blob/d0dbd19f75248adb4e56755a02c376a33f83f039/packages/components/tabs/src/tab-nav.tsx#L320

indeed fix this

same like the button component

**Comment by sxzz** [comment]
@tolking Just keep it as-is, this error will be fixed in v3. Please restore it ATM.

File: packages/components/tabs/src/tabs.tsx [hunk 1]
```diff
@@ -209,8 +209,8 @@ export default defineComponent({
             type={props.type}
             panes={Object.values(panes)}
             stretch={props.stretch}
-            onTabClick={handleTabClick}
-            onTabRemove={handleTabRemove}
+            onTab-click={handleTabClick}
+            onTab-remove={handleTabRemove}
```
Comment (sxzz) [inline packages/components/tabs/src/tabs.tsx:213]:
```suggestion
            onTabClick={handleTabClick}
            onTabRemove={handleTabRemove}
```

File: packages/components/tabs/src/tab-nav.tsx [hunk 2]
```diff
@@ -66,14 +54,21 @@ export const tabNavProps = buildProps({
   stretch: Boolean,
 } as const)
 
+export const tabNavEmits = {
+  'tab-click': (tab: TabsPaneContext, tabName: TabPanelName, ev: Event) =>
+    ev instanceof Event,
+  'tab-remove': (tab: TabsPaneContext, ev: Event) => ev instanceof Event,
```
Comment (sxzz) [inline packages/components/tabs/src/tab-nav.tsx:60]:
```suggestion
  tabClick: (tab: TabsPaneContext, tabName: TabPanelName, ev: Event) =>
    ev instanceof Event,
  tabRemove: (tab: TabsPaneContext, ev: Event) => ev instanceof Event,
```

File: packages/components/tabs/src/tabs.tsx (hunk 1 above)
Comment (tolking) [reply packages/components/tabs/src/tabs.tsx:213]: This will result in an error

Comment (sxzz) [reply packages/components/tabs/src/tabs.tsx:213]: OK I will try it later.

**Review by sxzz — APPROVED** [review]

**Pull request merged**